libexec/functions

```diff
@@ -310,13 +310,40 @@ parse_opts() {
 }
 
 
+compare_envs() {
+    # during bootstrap, if we are bootstrapping from a local image, compare the
+    # environment in the new image to that in the old image and alert the user
+    # of any changes
+    REMOVED=$(comm -23 ${SINGULARITY_STARTING_ENVIRONMENT} ${SINGULARITY_ENDING_ENVIRONMENT})
+    ADDED=$(comm -13 ${SINGULARITY_STARTING_ENVIRONMENT} ${SINGULARITY_ENDING_ENVIRONMENT})
+     
```

Comment:
`comm` is like `diff` but better.  It gives you 3 columns.  The first column has lines unique to file one, second column has files unique to file 2, and 3rd column has lines common to both.  The `-23` and `-13` omit the specified columns from the output.  I could just call it once and parse the output, but I opted to call it twice and not parse anything.

Also there is a caveat that the files must be sorted.  This thing was written by Stallman, so it's old (and GNU) and I think (hope) it should be available most everywhere.